packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -224,23 +254,80 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
 
   bool _debugAssertChildListLocked() => childManager.debugAssertChildListLocked();
 
+  /// Verify that the child list index is in strictly increasing order.
+  ///
+  /// This has no effect in release builds..
+  void _debugVerifyChildOrder(){
+    assert(() {
+      if (_debugChildIntegrityEnabled) {
+        if (firstChild != null) {
+          int index = indexOf(firstChild);
+          RenderBox child = childAfter(firstChild);
+          while (child != null) {
+            assert(indexOf(child) > index);
+            index = indexOf(child);
+            child = childAfter(child);
+          }
+        }
+      }
+      return true;
+    }());
+  }
+
   @override
   void insert(RenderBox child, { RenderBox after }) {
     assert(!_keepAliveBucket.containsValue(child));
     super.insert(child, after: after);
     assert(firstChild != null);
     assert(() {
-      int index = indexOf(firstChild);
-      RenderBox child = childAfter(firstChild);
-      while (child != null) {
-        assert(indexOf(child) > index);
-        index = indexOf(child);
-        child = childAfter(child);
-      }
+      _debugVerifyChildOrder();
       return true;
     }());
   }
 
+  @override
+  void move(RenderBox child, { RenderBox after }) {
+    // There are two scenarios:
+    //
+    // 1. The child is not keptAlive.
+    // The child is in childList maintained by ContainerRenderObjectMixin.
+    // We can call the super.move and update parentData with new slot.
```

Comment:
grammar nits:
`in the childList`
`call super.move` (no `the`)
`with the new slot`